File: java/src/main/java/com/a65apps/kalimullinilnazrafilovich/interactors/details/ContactDetailsModel.java
package com.a65apps.kalimullinilnazrafilovich.interactors.details;

import com.a65apps.kalimullinilnazrafilovich.entities.ContactDetailsInfo;

import io.reactivex.rxjava3.core.Single;

public class ContactDetailsModel implements ContactDetailsInteractor {
    private final transient ContactDetailsRepository contactDetailsRepository;

    public ContactDetailsModel(ContactDetailsRepository contactDetailsRepository) {
        this.contactDetailsRepository = contactDetailsRepository;
    }

    @Override
    public Single<ContactDetailsInfo> loadDetailsContact(String id) {
        return contactDetailsRepository.getDetailsContact(id);
    }

}

File: java/src/main/java/com/a65apps/kalimullinilnazrafilovich/interactors/details/ContactDetailsModel.kt
package com.a65apps.kalimullinilnazrafilovich.interactors.details

import com.a65apps.kalimullinilnazrafilovich.entities.ContactDetailsInfo
import kotlinx.coroutines.flow.Flow

class ContactDetailsModel(
        private val contactDetailsRepository: ContactDetailsRepository)
    : ContactDetailsInteractor {
    override fun loadDetailsContact(id: String): Flow<ContactDetailsInfo> {
        return contactDetailsRepository.getDetailsContact(id)
    }
}
